_articles/aws_sts_storage_locations.md

```diff
@@ -0,0 +1,182 @@
+                    stsEnabled=TRUE,
+                    bucket='nameofyourbucket',
+                    baseKey='nameofyourbasekey',
+                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting')
+destination <- synRestPOST('/storageLocation', body=toJSON(destination))
+
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- folderId
+
+projectDestination <- synRestPOST('/projectSettings', body=toJSON(projectDestination))
+```
+
+Once your STS Storage Location is set up on your Synapse folder, you can add files to Synapse in one of two ways:
+
+1. You can upload files through Synapse, either through [the Synapse website](https://www.synapse.org/) or through the client of your choice.
+2. If you plan to upload files directly to your S3 bucket, or if you already have files in your S3 bucket, you can add representations of those files to Synapse programmatically. Follow the steps at [Adding Files in Your S3 Bucket to Synapse](custom_storage_location.md#adding-files-in-your-s3-bucket-to-synapse).
+
+{% include note.html content="We recommend picking one of these methods and sticking to it and avoiding mixing methods." %}
+
+## Obtaining Temporary S3 Credentials
+
+Once your STS Storage Location is set up, you can use Synapse to request temporary AWS credentials with direct access to your data in S3. These temporary credentials are good for up to 12 hours.
+
+To get these temporary credentials, you may use our [REST interface](https://rest-docs.synapse.org/rest/GET/entity/id/sts.html).
+
+{% include note.html content="If your storage location is external storage, you may request read-only or read-write permissions. If that storage location is Synapse storage, you must request read-only permissions." %}
+
+Alternatively, you may use the below sample Java code:
+
+```java
+StsCredentials stsCredentials = synapseClient.getTemporaryCredentialsForEntity(folderEntityId, StsPermission.read_only);
+AWSCredentials awsCredentials = new BasicSessionCredentials(stsCredentials.getAccessKeyId(), stsCredentials.getSecretAccessKey(), stsCredentials.getSessionToken());
+AWSCredentialsProvider awsCredentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
+```
+
+## Migrating Existing Data
+
+### From an Existing S3 Bucket
+
+If you have existing data in an S3 bucket, either standalone data or data from a previous Synapse project, you can use [this sample code](https://github.com/Sage-Bionetworks/Synapse-Repository-Services/blob/develop/client/sample-code/src/main/java/org/sagebionetworks/sample/sts/MigrateS3Bucket.java) to migrate your S3 data to a new Synapse folder with an STS Storage Location.
```

Comment:
This particular sample code is for populating an External Storage Location with an existing S3 bucket. I wrote it in Java because I don't know R or Python.